app/bundles/SmsBundle/Config/config.php

```diff
@@ -195,12 +195,12 @@
         ],
     ],
     'parameters' => [
-        'sms_enabled'              => false,
-        'sms_username'             => null,
-        'sms_password'             => null,
-        'sms_sending_phone_number' => null,
-        'sms_frequency_number'     => 0,
-        'sms_frequency_time'       => 'DAY',
-        'sms_transport'            => 'mautic.sms.twilio.transport',
+        'sms_enabled'               => false,
+        'sms_username'              => null,
+        'sms_password'              => null,
+        'sms_messaging_service_sid' => null,
```

Comment:
@dadarya0 is this a BC break? The Twilio plugin will have to be re-configured after this change, right? Shouldn't there be a migration that would take the value of `sms_sending_phone_number` and set it to `sms_messaging_service_sid`?

Reply:
those two are not identical, different identifier. it is a BC break, it requires you to enter the new value in advanced configuration of the plugin. it's like routing key instead of phone number.

Reply:
When I look at the original documentation for the Twilio plugin:

https://docs.mautic.org/en/plugins/twilio

Reply:
@escopecz I have tested following thing :-
1. used Messaging Service SID value in `messagingServiceSid ` => Message delivered 
2. used phone number value in `messagingServiceSid` => not working , message not delivered

As per documentation also , We can either pass the Messaging Service SID in From parameter, or we can use its SID as the MessagingServiceSid parameter in API request. Either one will work the same but did not mentioned to pass phone number in MessagingServiceSid parameter.

https://support.twilio.com/hc/en-us/articles/223181308-Getting-started-with-Messaging-Services

So migration would not help, All customer will required to reconfigured twilio plugin with MessagingServiceSid

Reply:
It won't work even if users were using SID in the `sms_sending_phone_number` config param?

Reply:
As per documentation, it will work if user uses SID in from parameter.
In this case, no need to merge this PR just update SID in configuration instead phone number.

Reply:
I think this is a good change. It doesn't make sense to store SID into an option called "phone number". All I'm saying is that if they were using SID in the old config value then we can migrate it to the new config value and the users won't have to do any manual change.

Here is an example migration that modifies the config file:

https://github.com/mautic/mautic/blob/5.x/app/migrations/Version20230615101328.php

Reply:
@escopecz migration will not work

Account SID is not same like Messaging Service SID 

It will return `[HTTP 400] Unable to create record: The Messaging Service Sid ... is invalid.`

The only solution is to keep sending with phone number until Messaging Service SID  is set.

Reply:
Thanks @escopecz , now I got it , migration is needed for those user who has already started use of `messagingServiceSid` in our mautic twilio plugin configuration but I do not think that without any notification from mautic anyone would start use of `messagingServiceSid` in advance.

please let me know if still migration is needed then I will add write a migration for update integration configuration for twilio.

Today I found that [sms configuration params](https://github.com/mautic/mautic/blob/dc8dca4036d26a48b47fe4c827aa77af62b73eae/app/bundles/SmsBundle/Config/config.php#L198-L204) are useless as these configuration were used initially when we were using sms configuration from mautic global configuration but these are unused when we had started sms configuration from twilio plugin config.

In [this](https://github.com/mautic/mautic/commit/c33bfaed9b21a587ed64e1c69c74ab4d13263f97) commit, sms configuration was moved from global configuration to integration configuration but param was not removed.

Reply:
Now I get it. There are 2 different SIDs. So no migration is needed. Only the documentation mentioned in https://github.com/mautic/mautic/pull/12539#issuecomment-1629084702